src/main/java/io/confluent/kafkarest/resources/TopicsResource.java

```diff
@@ -74,6 +58,37 @@ public Topic getTopic(@PathParam("topic") String topicName) {
     return topic;
   }
 
+  @GET
+  @Path("/{topic}/partition/{partition}/messages")
```

Comment:
Can we move these to `PartitionsResource`? That would be the more natural location for them since these calls only make sense on a particular location. Also, right now it's inconsistent (this uses `/topcs/{topic}/partition/{partition}` but Partitions resource uses `/topics/{topic}/partitions/{partition}` so that the resource `/topics/{topic}/partitions` is clearly a collection rather than singular).